plugin/pkg/admission/allowunsafedelete/admission_test.go

```diff
@@ -0,0 +1,231 @@
+			},
+		},
+		{
+			name:           "feature enabled, delete, IgnoreStoreReadErrorWithClusterBreakingPotential is true, authorizer gives no opinion, forbid",
+			featureEnabled: true,
+			attr: newAttributes(attributes{
+				operation: admission.Delete,
+				operationOptions: &metav1.DeleteOptions{
+					IgnoreStoreReadErrorWithClusterBreakingPotential: ptr.To[bool](true),
+				},
+			}),
+			authz: &fakeAuthorizer{
+				decision: authorizer.DecisionNoOpinion,
+				reason:   "no opinion",
+			},
+			err: func(attr admission.Attributes) error {
+				return admission.NewForbidden(attr, fmt.Errorf("user not permitted to do %q, reason: %s", verb, "no opinion"))
+			},
+		},
+		{
+			name:           "feature enabled, delete, IgnoreStoreReadErrorWithClusterBreakingPotential is true, user has permission, admit",
+			featureEnabled: true,
+			attr: newAttributes(attributes{
+				operation: admission.Delete,
+				operationOptions: &metav1.DeleteOptions{
+					IgnoreStoreReadErrorWithClusterBreakingPotential: ptr.To[bool](true),
+				},
+				userInfo: &user.DefaultInfo{Name: "foo"},
+			}),
+			authz: &fakeAuthorizer{
+				decision: authorizer.DecisionAllow,
+				reason:   "permitted",
+			},
+		},
+		{
+			name:           "feature disabled, always admit",
+			featureEnabled: false,
+			attr: newAttributes(attributes{
+				operation: admission.Delete,
+				operationOptions: &metav1.DeleteOptions{
+					IgnoreStoreReadErrorWithClusterBreakingPotential: ptr.To[bool](true),
+				},
+				userInfo: &user.DefaultInfo{Name: "foo"},
+			}),
+			authz: &fakeAuthorizer{
+				err: fmt.Errorf("Authorize should not be invoked when feature is disabled"),
+			},
+		},
+	}
+
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, genericfeatures.AllowUnsafeMalformedObjectDeletion, test.featureEnabled)
+
+			admission := NewAllowUnsafeDelete()
+			admission.SetAuthorizer(test.authz)
+
+			var want error
+			if test.err != nil {
+				want = test.err(test.attr)
+			}
+
+			got := admission.Validate(context.Background(), test.attr, nil)
+			switch {
+			case want != nil:
+				if got == nil || want.Error() != got.Error() {
+					t.Errorf("expected error: %v, but got: %v", want, got)
+				}
+			default:
+				if got != nil {
+					t.Errorf("expected no error, but got: %v", got)
+				}
+			}
+		})
+	}
+}
+
+// attributes of interest for this test
+type attributes struct {
+	operation        admission.Operation
+	operationOptions runtime.Object
+	userInfo         user.Info
+}
+
+func newAttributes(attr attributes) admission.Attributes {
+	return admission.NewAttributesRecord(
+		nil,                           // this plugin should never inspect the object
+		nil,                           // old object
+		schema.GroupVersionKind{},     // this plugin should never inspect kind
+		"",                            // this plugin should never inspect namespace
```

Comment:
not true comment, right?

Reply:
the plugin simply passes it to the authorizer, i updated the comment